pkgs/games/papermc/default.nix

```diff
@@ -1,50 +1,69 @@
+{ lib, stdenvNoCC, fetchurl, jre, makeBinaryWrapper, ... }:
+
+let
+  generic =
+    { version, hash, ... }@ args:
+    stdenvNoCC.mkDerivation (args // {
+      pname = "papermc";
+      inherit version;
+
+      src =
+        let
+          mcVersion = lib.versions.pad 3 version;
+          buildNum = builtins.elemAt (lib.splitVersion version) 3;
+        in
+        fetchurl {
+          url = "https://papermc.io/api/v2/projects/paper/versions/${mcVersion}/builds/${buildNum}/downloads/paper-${mcVersion}-${buildNum}.jar";
```

Comment:
Would it be possible to change `version` to look like `1.20.2-234` instead of `1.20.2.234`?

That would eliminate the parsing logic, and I think it could be a easier for users (who may be confused by the extra decimal point)

Reply:
That sounds like a great idea!
My goal was, to firstly migrating the package 1:1 before changing it's logic.
One disadvantage I can think of: we would break setups, using some very specific overrides.

I will add it to my PR.

Reply:
Edit: why does it eliminate the parsing logic?
We still have to use build number and version separately in the URL.

Reply:
I implemented the suggested changes!
If you like it this way, I would squash the changes to a single commit.

Reply:
Oops, I didn't see that the version still needs to be split.

Do we think a lot of users are using specific overrides? I don't want to break users' setups if that's the case

Reply:
Puh, hard to say...
Generally speaking: pinning a specific version is quite important when hosting a Minecraft server.

That being said:
I don't think that is a big problem.
Because of the nix store (hash does not change), only new installs would fail. People using overrides would easily be able to look up what has changed.
As I demonstrated above, overrides get simplified by this PR too!

In my opinion:
This syntax makes it easier for new users.
People already having overrides will know how to change them, especially since they are now easier... (we could even add a comment with an example in theory?).